pageserver/src/tenant/timeline.rs

```diff
@@ -642,7 +642,13 @@ impl FlushLayerError {
     // When crossing from generic anyhow errors to this error type, we explicitly check
     // for timeline cancellation to avoid logging inoffensive shutdown errors as warn/err.
     fn from_anyhow(timeline: &Timeline, err: anyhow::Error) -> Self {
-        if timeline.cancel.is_cancelled() {
+        let cancelled = timeline.cancel.is_cancelled()
```

Comment:
Added in 98dadf854383f7d96cd6b30c87ba49b1b5a48d1e

Reply:
>This non-local reasoning is rather unfortunate, e

It's a transient state: to avoid having to convert the anyhow::Error uses down the stack right away, we have this as a cutoff point for "lower layers might give us anyhow::Error, but we won't propagate those as Other if we're shutting down".

End state should be that we have structured errors all the way down, and this function won't exist any more.

Reply:
Yeah, `timeline.cancel.is_cancelled()` is a bit sad, ideally we'd list all the possible errors that can occur and then convert them to their cancelled form.